There wasn't actual persistence, this is fixed

paymog opened the pull request:
Factories are made new for each controller. Services are not.
Persistence is now fixed using a service. Tests pass. We need
integration tests so that this could have been caught the first
time round.

I also did some misellaneous cleanup. Specifically, the controllers
for matched and matching states do not have URLs anymore.

paymog committed:
There wasn't actual persistence, this is fixed

Factories are made new for each controller. Services are not.
Persistence is now fixed using a service. Tests pass. We need
integration tests so that this could have been caught the first
time round.

I also did some misellaneous cleanup. Specifically, the controllers
for matched and matching states do not have URLs anymore.

kjrcda commented on webapp/public/javascripts/services.js:
```diff
@@ -0,0 +1,19 @@
+var lunchrServices = angular.module('lunchrServices', []);
+
+lunchrServices.service('authService', function () {
+    var user;
```
maybe default assign this to NULL?

paymog replied:
Will do.

kyleleung75 commented on webapp/public/javascripts/angular-app.js:
```diff
@@ -18,18 +18,18 @@ lunchrApp.config(['$stateProvider', '$urlRouterProvider', '$locationProvider',
                 controller: 'UserController'
             })
             .state('users.matching', {
-                url: '/matching',
```
why is it that the url field is now empty? same with line 26

paymog replied:
It didn't make sense to me to have a URL for this state. By having a URL it means that the user could potentially manually navigate to "/users/matching" even if they hadn't pressed the button to start being matched. Also, the semantics just seemed off having the URL be modified to start being matched and modified again once a match is found.

kyleleung75 replied:
Oh that makes sense, cool.

kjrcda commented:
looks good

paymog committed:
Current user didn't persist through reloads. Use cookies to fix.

Use ngCookies to persist the current user through reloads and navigating
away from the app.

kyleleung75 commented:
It looks good now

paymog commented:
Lol finally

On Tue, Mar 10, 2015, 12:18 AM kyleleung75 notifications@github.com wrote:

> It looks good now
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/paymog/Lunchr/pull/37#issuecomment-77997434.
